superset-frontend/src/components/AlteredSliceTag.jsx

```diff
@@ -65,7 +65,18 @@ export default class AlteredSliceTag extends React.Component {
       return;
     }
     const diffs = this.getDiffs(newProps);
-    this.setState({ diffs, hasDiffs: !isEmpty(diffs) });
+    this.setState(prevState => ({
+      rows: this.getRowsFromDiffs(diffs, prevState.controlsMap),
+      hasDiffs: !isEmpty(diffs),
+    }));
+  }
+
+  getRowsFromDiffs(diffs, controlsMap = this.state.controlsMap) {
```

Comment:
is this default arg only used in the test? If so, it may be simpler to just use a static value in the test or just spy on this method and see if it got called. I'm not sure if testing against the value from this method adds much value in the test if the test is for checking that `UNSAFE_componentWillReceiveProps` called this action. If you are looking to see what the value is from this method, then making a separate unit test just for this method would be better, imo.

Reply:
I agree, that test made little sense. I changed the logic so that we spy on `getRowsFromDiffs`  to check if it was called and if `state.rows` equals a mocked return value from `getRowsFromDiffs`.
Is that what you meant?

Reply:
yeah, that looks good!